agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -3266,6 +3287,80 @@ bool backhaul_manager::handle_1905_autoconfiguration_response(ieee1905_1::CmduMe
     return true;
 }
 
+bool backhaul_manager::handle_1905_beacon_metrics_query(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                        const std::string &src_mac,
+                                                        Socket *&forward_to)
+{
+    LOG(DEBUG) << "now going to handle BEACON METRICS QUERY";
+
+    // extract the desired STA mac
+    auto tlvBeaconMetricsQuery = cmdu_rx.getClass<wfa_map::tlvBeaconMetricsQuery>();
+    if (!tlvBeaconMetricsQuery) {
+        LOG(ERROR) << "handle_1905_beacon_metrics_query should handle only tlvBeaconMetrics, but "
+                      "got something else: 0x"
+                   << std::hex << (uint16_t)cmdu_rx.getMessageType();
+        return false;
+    }
+
+    const sMacAddr &requested_sta_mac = tlvBeaconMetricsQuery->associated_sta_mac();
+    LOG(DEBUG) << "the requested STA mac is: " << requested_sta_mac;
+
+    const auto radio = get_sta_radio(requested_sta_mac);
+
+    // build ACK message CMDU
+    const auto mid      = cmdu_rx.getMessageId();
+    auto cmdu_tx_header = cmdu_tx.create(mid, ieee1905_1::eMessageType::ACK_MESSAGE);
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "cmdu creation of type ACK_MESSAGE, has failed";
+        return false;
+    }
+
+    if (!radio) {
+        LOG(DEBUG) << "STA with MAC [" << requested_sta_mac
+                   << "] is not associated with any BSS operated by the agent";
+
+        // add an Error Code TLV
+        auto error_code_tlv = cmdu_tx.addClass<wfa_map::tlvErrorCode>();
+        if (!error_code_tlv) {
+            LOG(ERROR) << "addClass wfa_map::tlvErrorCode has failed";
+            return false;
+        }
+
+        error_code_tlv->reason_code() =
+            wfa_map::tlvErrorCode::STA_NOT_ASSOCIATED_WITH_ANY_BSS_OPERATED_BY_THE_AGENT;
+
+        error_code_tlv->sta_mac() = requested_sta_mac;
+
+        // report the error
+        std::stringstream errorSS;
+        auto error_tlv = cmdu_tx.getClass<wfa_map::tlvErrorCode>();
+        if (error_tlv) {
+            errorSS << "0x" << error_tlv->reason_code();
+        } else {
+            errorSS << "note: error constructing the error itself";
+        }
+
+        LOG(DEBUG) << "sending ACK message to the originator with an error, mid: " << std::hex
+                   << int(mid) << " tlv error code: " << errorSS.str();
```

Comment:
Why do you insist on making simple things complicated?
```suggestion
        LOG(DEBUG) << "sending ACK message to the originator with an error, mid: " << std::hex
                   << int(mid) << " tlv error code: STA_NOT_ASSOCIATED_WITH_ANY_BSS_OPERATED_BY_THE_AGENT";
```

But even that is redundant - you could just add to the log message at the beginning of this condition that you're going to send an ack with error code. Or do you really want a log message for every line of code?